beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -279,28 +299,76 @@ func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeacon
 		newBlks = append(newBlks, blk)
 	}
 	if len(newBlks) == 0 {
-		delete(s.slotToPendingBlocks, slot)
-		return
+		s.slotToPendingBlocks.Delete(slotToCacheKey(slot))
+		return nil
+	}
+
+	// Decrease exp itme in proportion to how many blocks are still in the cache for slot key.
+	d := time.Duration(pendingBlockExpTime.Nanoseconds() / int64(len(newBlks)))
```

Comment:
```suggestion
	d := pendingBlockExpTime / int64(len(newBlks))
```
its already in nanoseconds

Reply:
fyi this wouldn't work. One is `time.Duration` and one is `int64`. I ended up changing it to 

`pendingBlockExpTime / time.Duration(len(newBlks))`